sys/net/link_layer/l2util/l2util.c

```diff
@@ -162,6 +195,22 @@ int l2util_ipv6_iid_to_addr(int dev_type, const eui64_t *iid, uint8_t *addr)
             addr[0] ^= 0x02;
             return sizeof(eui64_t);
 #endif  /* defined(MODULE_NETDEV_IEEE802154) || defined(MODULE_XBEE) */
+#if defined (MODULE_NRF24L01P)
+        case NETDEV_TYPE_NRF24L01P:;
+            uint8_t i = 0;
+            uint8_t j = 0;
+            while (i < 3 && iid->uint8[i]) {
+                addr[j++] = iid->uint8[i++];
+            }
+            i = 5;
+            while (i < 8 && !iid->uint8[i]) {
+                i++;
+            }
+            while (i < 8) {
+                addr[j++] = iid->uint8[i++];
+            }
+            return j;
```

Comment:
This could potentially leave part of the address uninitialized. How about just using:

```C
/* EUI64 is 8 bytes long, but nRF24l01+ addresses are only 5. So take the first 5 bytes
 * and xor the remaining bytes in */
memcpy(addr, iid->uint8, 5);
for (unsigned i = 5; i < sizeof(iid->uint8[i]); i++) {
    addr[i - 5] ^= iid->uint8[i];
}
```

The good thing about modern compilers is that they know the semantics of `memset()`, `memcpy()`, `memmove()`. So they will just emit equivalent but more efficient code instead of the actual function call, if this is possible.

Reply:
Ok, I will try that.

Reply:
A l2-address 3, 4 or 5 bytes long.
Is it a problem if two l2-addresses map to the same EUI64, if the interfaces are not on the same system? If the EUI64 must be unique only system wide, it would be sufficient to convert
a l2-address to a EUI64 like this: `XX:XX:03:02:01 --> 00:00:00:ff:fe:03:02:01` because the LSByte must be different for all pipes.

"The LSByte must be unique for all six pipes." (datasheet p. 38)

However, if a user has more than 1 `nrf24l01+` connected to the board, he must make sure that all addresses differ in the LSByte

Since the first byte of the EUI64 will always be zero, the EUI64 will always indicate local uniqueness.

Reply:
Nevermind, I cannot generate a distinct l2-address with this.

Reply:
I think I have to encode in the EUI64, whether the l2-address has 3, 4 or 5 bytes.
Let´s say the first 2 bits of the MSByte of the EUI64 encode the l2-address length:
00xxxxxx or 11xxxxxx -> 3 bytes
01xxxxxx -> 4 bytes
10xxxxxx -> 5 bytes